src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt

```diff
@@ -24,28 +23,164 @@ class ThemePatch : ResourcePatch {
+            /**
+             * Create a new node element of type 'item'
+             * with the attribute: 'name', and its value based on the field 'filePathIndex':
+             *
+             *      '/values/styles.xml' = 'splashScreenColor'
+             *      '/values-v31/styles.xml' = 'android:windowSplashScreenBackground'
+             *
+             * @field filePathIndex The current styles.xml path, indexed by the method 'forEachIndexed()'
+             */
+            file.createElement("item").apply {
+                setAttribute(
+                    "name",
+                    when (filePathIndex) {
+                        0 -> splashScreenColorAttributeName
+                        1 -> "android:windowSplashScreenBackground"
+                        else -> "null"
+                    }
+                )
 
-                node.textContent = when (node.getAttribute("name")) {
-                    "yt_black0", "yt_black1", "yt_black1_opacity95", "yt_black1_opacity98", "yt_black2", "yt_black3",
-                    "yt_black4", "yt_status_bar_background_dark", "material_grey_850" -> darkThemeBackgroundColor
+                /**
+                 * Fill the value of created node element with the field 'lightThemeBackgroundColor' or 'darkThemeBackgroundColor'
+                 * only if the name of 'indexedNode' is equal to the field 'nodeAttributeName'
+                 */
+                appendChild(
+                    file.createTextNode(
+                        when (filePathIndex) {
+                            0 -> when (nodeAttributeName) {
+                                "Base.Theme.YouTube.Launcher.Dark" -> darkThemeBackgroundColor
+                                "Base.Theme.YouTube.Launcher.Light" -> lightThemeBackgroundColor
+                                else -> "null"
+                            }
+                            1 -> when (nodeAttributeName) {
+                                "Base.Theme.YouTube.Launcher" -> "?attr/$splashScreenColorAttributeName"
+                                else -> "null"
+                            }
+                            else -> "null"
+                        }
+                    )
+                )
 
-                    "yt_white1", "yt_white1_opacity95", "yt_white1_opacity98", "yt_white2", "yt_white3", "yt_white4",
-                     -> lightThemeBackgroundColor
+                /**
+                 * Add the created node element after 'indexedNode', into xml file
+                 * only if the value of new node element is not equal to 'null'
+                 */
+                if (this.textContent != "null")
+                    indexedNode.appendChild(this)
+            }
+        }
+        fun changeSplashScreenBackgroundColor(rootNode: Element) {
+            /**
+             * Check if a node element with attribute 'android:drawable' exists
+             * then change its value to '?attr/splashScreenColor'
+             */
```

Comment:
This comment is useless as it simply duplicates code as prescribed in rule 1 [here](https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments).

Reply:
So in this case I should remove the duplicate comments, or deduplicate the code together with comments?

Reply:
No, a comment according to [this](https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments) should be added. I also gave an example [here](https://github.com/revanced/revanced-patches/pull/1315#discussion_r1057997631).